module-system/module-implementations/sov-evm/src/call.rs

```diff
@@ -29,53 +30,75 @@ impl<C: sov_modules_api::Context> Evm<C> {
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
         let evm_tx_recovered: TransactionSignedEcRecovered = tx.try_into()?;
+        let block_env = self
+            .pending_block
+            .get(working_set)
+            .expect("Pending block must be set");
 
-        let block_env = self.pending_block.get(working_set).unwrap_or_default();
-        let cfg = self.cfg.get(working_set).unwrap_or_default();
+        let cfg = self.cfg.get(working_set).expect("Evm config must be set");
         let cfg_env = get_cfg_env(&block_env, cfg, None);
 
-        let hash = evm_tx_recovered.hash();
-
         let evm_db: EvmDb<'_, C> = self.get_db(working_set);
+        let result = executor::execute_tx(evm_db, &block_env, &evm_tx_recovered, cfg_env);
 
-        // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/505
-        let result = executor::execute_tx(evm_db, block_env, &evm_tx_recovered, cfg_env).unwrap();
+        let receipt = match result {
+            Ok(result) => {
+                let logs: Vec<_> = result.logs().into_iter().map(into_reth_log).collect();
 
-        let from = evm_tx_recovered.signer();
-        let to = evm_tx_recovered.to();
-        let transaction = reth_rpc_types::Transaction::from_recovered(evm_tx_recovered);
+                let mut accessory_state = working_set.accessory_state();
+                let tx_count = self.pending_transactions.len(&mut accessory_state);
 
-        self.pending_transactions
-            .push(&transaction, &mut working_set.accessory_state());
+                let previous_transaction: Option<PendingTransaction> = if tx_count == 0 {
+                    None
+                } else {
+                    Some(
+                        self.pending_transactions
```

Comment:
It looks like we should add `last ` method to `AccessoryStateVec & StateVec` and use it here.  Can be done in another PR.